src/api/app/models/project.rb

```diff
@@ -532,6 +528,20 @@ def fill_repo_cache
   def update_one_repository_add_pathes(repo)
     current_repo = self.repositories.find_by_name(repo['name'])
 
+    # sync download on demand config
+    hasharray=[]
+    repo.elements('download') { |d|
+      hash = { arch: d['arch'], url: d['url'], repotype: d['repotype'],
+               archfilter: d['archfilter'], pubkey: d['pubkey']
+             }
+      if d['master']
+        hash[:masterurl] = d['master']['url']
+        hash[:mastersslfingerprint] = d['master']['sslfingerprint']
+      end
+      hasharray << hash
+    }
+    sync_hash_with_model(DownloadRepository, current_repo.download_repositories, hasharray)
```

Comment:
there is no function in activerecord which is handling a set of records via a given key to my knowledge. (modifying, adding and removing all entries for a specified identifier)